servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/requestExtensions.kt

```diff
@@ -34,3 +34,5 @@ fun GraphQLRequest.toExecutionInput(graphQLContext: Any? = null, dataLoaderRegis
         }
         .dataLoaderRegistry(dataLoaderRegistry ?: DataLoaderRegistry())
         .build()
+
+fun GraphQLRequest.isMutation(): Boolean = query.contains("mutation ")
```

Comment:
This would pass if someone just had a field called `mutation` as well....

Are we going to have to parse the request?

Reply:
Technically it would also catch any `mutation` field suffix , e.g. `permutation`. I don't think it matters much -> we care if it is NOT a mutation so we can parallelize the calls, otherwise it will fallback to current sequential behavior

Reply:
did an investigation to see if its possible to to "cache" a `parsedDocument`, however graphql-java performs parsing and validation (syntactical and semantical executions) together, for this use case we only care about parsing to now if the query inside the document is a `mutation`,

is still possible to achieve this by implementing the  [PreparsedDocumentProvider](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/execution/preparsed/PreparsedDocumentProvider.java), and parseAndValidate the query there, and then store the result in the ExecutionInput local context, but that would imply duplicating code from the [GraphQL API ](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/GraphQL.java)

specifically adding this part 
https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/GraphQL.java#L541


i think is not worth it, and we can just do the `contains` check, and log when a query has "mutation" as a fielf suffix .